File: src/B_1547/B_1547.java
package B_1547;

import java.io.*;
import java.util.StringTokenizer;

public class B_1547 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        boolean[] ball = new boolean[4];
        StringTokenizer st;
        ball[1] = true;
        boolean temp = false;
        for (int i = 0; i < t; ++i) {
            st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            temp = ball[x];
            ball[x] = ball[y];
            ball[y] = temp;


        }
        for (int i = 1; i <= 3; ++i) {
            if(ball[i]) System.out.println(i);
        }
    }
}

File: src/B_1547/B_1547.kt
package B_1547

import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val t = readLine().toInt()
    val ball = BooleanArray(4)
    ball[1] = true
    var temp = false
    repeat(t){
        val a: IntArray = readLine().split(" ").toTypedArray().map {s -> s.toInt()}.toIntArray()
        temp = ball[a[0]]
        ball[a[0]] = ball[a[1]]
        ball[a[1]] = temp

    }
    for (i in ball.indices)
        if (ball[i]){
            println(i)
        }
}